### PR TITLE
Add Harvester cloud provider and csi

### PR DESCRIPTION
**Pull request by yaocw2020**
Support Harvester cloud provider and CSI in RKE2

Related Issue:
https://github.com/rancher/rancher/issues/34438
https://github.com/harvester/harvester/issues/1019

### CODE REVIEW

**Review by gitlawr — DISMISSED** [review]
Changes LGTM. Please resolve the conflicts.

**Comment by yaocw2020** [comment]
> Changes LGTM. Please resolve the conflicts.

Done

**Comment by guangbochen** [comment]
@yaocw2020 ditto, can u rebase it, thanks

**Review by brandond — DISMISSED** [review]

**Comment by brandond** [comment]
This looks good, but should not be merged until the rke2 version in KDM is bumped to a release that contains these charts.

**Comment by cjellick** [comment]
In addition to what @brandond said (needs to wait for an RKE2 version bump), @ibuildthecloud @deniseschannon and I need to have a discussion about how multiple versions of rke2 should be supported.

Denise and I aren't sure yet if we can just "replace" the existing rke2 version with a new one, because now this data also drives the "schema" that is exposed in the API and ultimately the UI.

**Comment by cjellick** [comment]
So, @yaocw2020 you need to leave the existing 1.21.4 entry as-is and create a new entry for 1.21.5.

You cannot modify the existing `charts-v1` and `agentArgs-v1` yaml data structures. They need to stay as-is for 1.21.4. You need to create new data structures, but @ibuildthecloud told me that you do not need to copy and paste it whole sale. YAML has tricks for doings things like referencing a structure and replacing specific fields on it. I don't know how to do that, but hoping you can figure it out.

**Comment by brandond** [comment]
I can update.

**Comment by brandond** [comment]
@cjellick how's that look

File: channels-rke2.yaml [hunk 1]
```diff
@@ -116,3 +116,29 @@ releases:
       rancher-vsphere-csi:
         repo: rancher-charts
         version: 100.0.0
+  - version: v1.21.5+rke2r1
```
Comment (cjellick) [inline channels-rke2.yaml]: This now needs updated to `v1.21.4-rc1+rke2r4`

and go generate reran

**Comment by brandond** [comment]
Updated to v1.21.4-rc1+rke2r4

**Review by cjellick — APPROVED** [review]

**Pull request merged**